### PR TITLE
Sidekiq

### PR DESCRIPTION
**Pull request by guragt**
1. Added gems: sidekiq, sidekiq-cron
2. Added job for sending email with user's statistics
Screenshot 2021-04-22 at 11 52 55

### CODE REVIEW

File: app/models/relationship.rb [hunk 1]
```diff
@@ -8,11 +8,11 @@ class Relationship < ApplicationRecord
...
   validates :status, presence: true
 
-  after_create :send_relationship_notification, unless: -> { Rails.env.test? }
+  after_create :send_relationship_notification
 
   private
 
   def send_relationship_notification
-    RelationshipMailer.new_follower(self).deliver_later
+    SendNewRelationshipEmailJob.perform_later(id)
```
Comment (boikonataliii) [inline app/models/relationship.rb]: You don't need to put this into job. 
Just make sure that your config is correct :  https://gist.github.com/maxivak/690e6c353f65a86a4af9

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class UserMailer < ApplicationMailer
+  def user_statistics_email
+    @report_period = 1.week.ago.all_week
+
+    @total_users_count = User.all.count
```
Comment (boikonataliii) [inline app/mailers/user_mailer.rb]: you don't need .all 
` User.count`

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    @new_users_count = User.with_deleted.where(created_at: @report_period).count
+    @deleted_users_count = User.with_deleted.where(deleted_at: @report_period).count
+
+    @admins_email = Administrator.all.pluck(:email)
```
Comment (boikonataliii) [inline app/mailers/user_mailer.rb]: `Administrator.pluck(:email)`

File: app/jobs/send_new_relationship_email_job.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class SendNewRelationshipEmailJob < ApplicationJob
+  queue_as :mailers
+
+  def perform(relationship_id)
+    relationship = Relationship.find_by(id: relationship_id)
+    return if relationship.nil?
+
+    RelationshipMailer.new_follower(relationship).deliver_now
```
Comment (boikonataliii) [inline app/jobs/send_new_relationship_email_job.rb]: use .deliver_later

File: .travis.yml [hunk 5]
```diff
@@ -3,9 +3,9 @@ cache:
   bundler: true
 language: ruby
 install:
-  - nvm install node
+  - travis_retry nvm install 13.7.0
```
Comment (boikonataliii) [inline .travis.yml:6]: do you still need this?

Comment (guragt) [reply .travis.yml:6]: Yes

**Review by boikonataliii — APPROVED** [review]

**Pull request merged**